urlddedf61e88_dwendelen_CG2_1_tgz-pJ8-RendProjectJ8/src/Vectormath/Vector3.java

```diff
@@ -0,0 +1,306 @@
+    if (tmp < 0.0) tmp = 0.0;
+    return(tmp);
+  }
+
+  /** Compute the distance between two points
+   * @param a 1st point
+   * @param b 2nd point
+   * @return Cartesian distance between the two points
+   */
+  public static double distance(Vector3 a,Vector3 b) {
+    double d;
+    d = Math.sqrt((a.data[0]-b.data[0])*(a.data[0]-b.data[0])+(a.data[1]-b.data[1])*(a.data[1]-b.data[1])+(a.data[2]-b.data[2])*(a.data[2]-b.data[2]));
+    return(d);
+  }
+
+  /** Compute linear combination of vectors and scalars
+   * result = a.x + b.y, with x,y vectors and a,b scalars
+   * @param a scalar for x
+   * @param x vector weighted with a
+   * @param b scalar for y
+   * @param y vector weighted with b
+   * @return resulting vector (new object)
+   */
+  public static Vector3 linearCombination(double a,Vector3 x,double b,Vector3 y) {
+    Vector3 result = new Vector3();
+    result.data[0] = a*x.data[0] + b*y.data[0];
+    result.data[1] = a*x.data[1] + b*y.data[1];
+    result.data[2] = a*x.data[2] + b*y.data[2];
+    return(result);
+  }
+
+  /** Compute linear combination of vectors and scalars
+   * result = a.x + b.y + c.z, with x,y,z vectors and a,b,c scalars
+   * @param a scalar for x
+   * @param x vector weighted with a
+   * @param b scalar for y
+   * @param y vector weighted with b
+   * @param c scalar for z
+   * @param z vector weighted with c
+   * @return resulting vector (new object)
+   */
+  public static Vector3 linearCombination(double a,Vector3 x,double b,Vector3 y,double c,Vector3 z) {
+    Vector3 result = new Vector3();
+    result.data[0] = a*x.data[0] + b*y.data[0] + c*z.data[0];
+    result.data[1] = a*x.data[1] + b*y.data[1] + c*z.data[1];
+    result.data[2] = a*x.data[2] + b*y.data[2] + c*z.data[2];
+    return(result);
+  }
+
+   /** Compute linear combination of vectors and scalars
+   * result = x + b.y + c.z, with x,y,z vectors and a,b,c scalars
+   * @param x vector
+   * @param b scalar for y
+   * @param y vector weighted with b
+   * @param c scalar for z
+   * @param z vector weighted with c
+   * @return resulting vector (new object)
+   */
+  public static Vector3 linearCombination(Vector3 x,double b,Vector3 y,double c,Vector3 z) {
+    Vector3 result = new Vector3();
+    result.data[0] = x.data[0] + b*y.data[0] + c*z.data[0];
+    result.data[1] = x.data[1] + b*y.data[1] + c*z.data[1];
+    result.data[2] = x.data[2] + b*y.data[2] + c*z.data[2];
+    return(result);
+  }
+  
+  /** Compute linear combination of vectors and scalars
+   * result = x + b.y, with x,y vectors and b scalar
+   * @param x vector
+   * @param b scalar for y
+   * @param y vector weighted with b
+   * @return resulting vector (new object)
+   */
+  public static Vector3 linearCombination(Vector3 x,double b,Vector3 y) {
+    Vector3 result = new Vector3();
+    result.data[0] = x.data[0] + b*y.data[0];
+    result.data[1] = x.data[1] + b*y.data[1];
+    result.data[2] = x.data[2] + b*y.data[2];
+    return(result);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 276:280, 291:295.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.